tests/src/common/test_case.rs

```diff
@@ -288,9 +291,197 @@ impl<Dispatcher, Treasury, Profit, Leaser, Lpp, Oracle, TimeAlarms>
     }
 }
 
-pub struct TestCase<Dispatcher, Treasury, Profit, Leaser, Lpp, Oracle, TimeAlarms> {
-    pub app: MockApp,
-    pub message_receiver: WrappedCustomMessageReceiver,
+#[must_use]
+pub(crate) struct WrappedApp {
+    app: MockApp,
+    message_receiver: WrappedCustomMessageReceiver,
+}
+
+impl WrappedApp {
+    pub const fn new(app: MockApp, message_receiver: WrappedCustomMessageReceiver) -> Self {
+        Self {
+            app,
+            message_receiver,
+        }
+    }
+
+    #[must_use]
+    #[inline]
```

Comment:
let's not clutter the code with useless compiler directives

Reply:
Those are the tests, so I am not quite sure why would the `#[inline]` matter, but I'd like to keep the `#[must_use]` as there are way too many places where the results were just implicitly ignored. That would at least make it clear, that we specifically say "Yes, I don't care about this value".

Reply:
don't or do care?

Reply:
The secondary idea behind this PR is to make the "I don't care about this value" explicit, thus the `#[must_use]`.